modules/programs/hexchat.nix

```
@@ -0,0 +1,248 @@
+{ config, pkgs, lib, ... }:
+with builtins;
+with lib;
+let cfg = config.programs.hexchat;
+in {
```

Here's the general pattern I recommend using:

```suggestion
{ config, pkgs, lib, ... }:

with builtins;
with lib;

let
  cfg = config.programs.hexchat;
  
  modChannelOption = with lib.types; {
  };
  
  channelOption = with lib.types; {
  };
  
  # Helpers
  transformChannels =
in
{
  options.programs.hexchat = {
    enable = mkEnableOption ...;
    
    channels = mkOption ...;
    
    settings = mkOption ...;
    
    theme = mkOption ...;
  };
  
  config = mkIf cfg.enable {
  };
}
```